### PR TITLE
Add multichannel support

### PR DESCRIPTION
**Pull request by DevNambi**
This branch adds multichannel support to the CellProfiler workflow. It does this almost entirely in the ParseCsv process, which now has 2 sets of outputs, 'shards' and 'csv'. Each shard gets both a 'shard' and a 'csv' file.

'Shards' are used *only by Nextflow*, they map shard numbers to the files to load for each. 'csv' outputs are CSV files used by the CellProfiler process, filtered to the files/groups specific to that shard.

As part of this refactor, the parse_csv template was refactored to use pandas.

### CODE REVIEW

File: main.nf [hunk 1]
```diff
@@ -8,9 +8,6 @@ params.help = false
 params.input_h5 = false
```
Comment (sminot) [inline main.nf]: I would prefer moving all default parameter values to `nextflow.config` to make it as easy as possible to check in a single place when configuring workflow execution in a wrapper script (or in the portal)

File: nextflow.config [hunk 2]
```diff
@@ -17,10 +17,68 @@ params {
...
+                cpus = 1
+                memory = 1.GB
+            }
+            withLabel: cpu_verylarge {
+                cpus = 1
+                memory = 1.GB
+            }
+            errorStrategy = 'retry'
+            maxRetries = 5
+            maxForks = 10
```
Comment (sminot) [inline nextflow.config]: I don't think we should limit the rate of execution to 10 concurrent processes by default

**Review by sminot — APPROVED** [review]
I think the only non-stylistic change that I caught was the `maxForks` parameter. Apart from that, it looks great and I'm glad that you got the pandas refactor in!

File: nextflow.config (hunk 2 above)
Comment (DevNambi) [reply nextflow.config]: Ah, sorry. This was a remnant from some testing; I'll fix it.

File: main.nf (hunk 1 above)
Comment (DevNambi) [reply main.nf]: Oh interesting. I'd done this with all of the non 'false' parameters, but I can certainly do it with these as well.

**Pull request merged**